src/main/java/com/puppycrawl/tools/checkstyle/utils/ModuleReflectionUtils.java

```diff
@@ -76,17 +78,36 @@ public static boolean isCheckstyleModule(Class<?> clazz) {
                     || isFilterModule(clazz)
                     || isFileFilterModule(clazz)
                     || isTreeWalkerFilterModule(clazz)
+                    || isAuditListener(clazz)
```

Comment:
@romani This is the only way to intercept detailed communications in Checkstyle.

I wrote a custom listener and expanded what communications it receives to gather and print metrics on it's execution time and number of times areas were called to produce reports and examine slow down areas, like modules and javadoc. You have seen these reports before. FileSets and Checks cannot receive the detailed information that a listener can as it is the most global thing we have in Checkstyle.

This is my branch: https://github.com/rnveach/checkstyle/commits/more_audits

I also have a private listener which is basically an implementation of https://github.com/checkstyle/checkstyle/issues/3242 , where on `auditFinished` it examines the list of defined suppressions and prints violations on any that were not used.
I have to use a listener as it is the only customizable area that is called at the end once all other modules, filters, and such are done.